packages/playout-gateway/src/tsrHandler.ts

```diff
@@ -1013,75 +837,18 @@ export class TSRHandler {
 		process.stderr.on('data', (data) => this.logger.info(data.toString()))
 		process.on('close', () => process.removeAllListeners())
 	}
-	private async _removeDevice(deviceId: string): Promise<DeviceActionResult> {
-		let success = false
-		if (this._coreTsrHandlers[deviceId]) {
-			try {
-				await this._coreTsrHandlers[deviceId].dispose()
-				this.logger.debug('Disposed device ' + deviceId)
-				success = true
-			} catch (error) {
-				this.logger.error(`Error when removing device "${deviceId}"`, { data: error })
-			}
-		}
-		delete this._coreTsrHandlers[deviceId]
 
-		return {
-			deviceId,
-			action: DeviceAction.REMOVE,
-			success,
-		}
-	}
 	private _triggerupdateExpectedPlayoutItems() {
 		if (!this._initialized) return
 		if (this._triggerupdateExpectedPlayoutItemsTimeout) {
 			clearTimeout(this._triggerupdateExpectedPlayoutItemsTimeout)
 		}
 		this._triggerupdateExpectedPlayoutItemsTimeout = setTimeout(() => {
-			this._updateExpectedPlayoutItems().catch((error) =>
-				this.logger.error('Error in _updateExpectedPlayoutItems', { data: error })
-			)
-		}, 200)
-	}
-	private async _updateExpectedPlayoutItems() {
-		const expectedPlayoutItems = this._coreHandler.core.getCollection('expectedPlayoutItems')
-		const peripheralDevice = this._getPeripheralDevice()
-
-		const expectedItems = expectedPlayoutItems.find({
-			studioId: peripheralDevice.studioId,
-		})
-		const rundowns = _.indexBy(
-			this._coreHandler.core.getCollection('rundowns').find({
-				studioId: peripheralDevice.studioId,
-			}),
-			'_id'
-		)
-
-		await Promise.all(
-			_.map(this.tsr.getDevices(), async (container) => {
-				if (!(await container.device.supportsExpectedPlayoutItems)) {
-					return
-				}
-				await container.device.handleExpectedPlayoutItems(
-					_.map(
-						_.filter(
-							expectedItems,
-							(item) => item.deviceSubType === container.deviceType
-							// TODO: implement item.deviceId === container.deviceId
-						),
-						(item): ExpectedPlayoutItem => {
-							const itemContent: ExpectedPlayoutItemContent = item.content
-							return {
-								...itemContent,
-								rundownId: item.rundownId,
-								playlistId: item.rundownId && rundowns[item.rundownId]?.playlistId,
-								baseline: item.baseline,
-							}
-						}
-					)
-				)
+			const { groupedExpectedItems, rundowns } = this.getExpectedPlayoutItems()
+			this.tsr.getDevices().forEach((device) => {
+				this.enqueueExpectedPlayoutItemsUpdate(groupedExpectedItems, rundowns, device.deviceId)
 			})
-		)
+		}, 200)
```

Comment:
Perhaps this should be extracted into a constant.

Reply:
Extracted